types/src/reference_tests.rs

```diff
@@ -77,6 +84,68 @@ async fn reference_payload() -> Payload {
         .0
 }
 
+async fn reference_ns_proof_legacy() -> ADVZNamespaceProofQueryData {
+    let payload = reference_payload().await;
+    let ns_index = payload
+        .ns_table()
+        .find_ns_id(&(REFERENCE_NAMESPACE_ID.into()))
+        .unwrap();
+    let enc = payload.encode();
+    let mut scheme = advz_scheme(10);
+    let disperse = VidScheme::disperse(&mut scheme, &enc).unwrap();
+
+    let ns_proof = NsProof::new(&payload, &ns_index, &VidCommon::V0(disperse.common)).unwrap();
```

Comment:
we may directly use `ADVZNsProof::new()` here, so you don't have to unwrap down below.

Reply:
https://github.com/EspressoSystems/espresso-network/pull/3121/commits/748073b1e1bf8bb31d1103be21621086319d77f7